### DISCUSSION TITLE
oAuth support to send emails via Microsoft 365 and Gmail

### DISCUSSION THREAD

**Discussion by venturini-dev** [craftcms/cms#15791] [Ideas] [answered]
With Microsoft discontinuing support for SMTP auth in September 2025, many clients using 365 as a mail server are moving towards oAuth to authenticate external apps.
In many cases we had to provide an external SMTP service just to send emails from a Craft CMS site.

It's never nice to have to answer that Craft cannot connect to a mailbox on 365 via oAuth.
This can also be useful for Gmail accounts.

Can you consider adding a transport type that involves oAuth authentication for sending emails via 365 and Gmail?

**Comment by recursivemonk** [comment]
I think I have just run into this issue myself.  When I use Wordpress sites there are plugins available that allow the site to post directly into Gmail using Oauth2.

If you find a solution to this, that would be great.  It looks like the only workaround might be to use App Passwords when it comes to Gmail accounts, but this is less than ideal compared to Oauth2.

I've looked for someone who has a 3rd party plug-in for Craft that might solve this issue, but I've yet to identify anything.

**Comment by brandonkelly** [comment]
We’re still looking into this, but it looks like app passwords are unaffected by Gmail’s upcoming changes per [this support article](https://apps.google.com/supportwidget/articlehome?article_url=https%3A%2F%2Fsupport.google.com%2Fa%2Fanswer%2F14114704&assistant_id=generic-unu&product_context=14114704&product_name=UnuFlow&trigger_context=a). Those are working in Craft 4 and 5 in our testing.

**Comment by TheFunkyMonk** [comment]
Is this still being looked into? We have quite a few clients that are using O365 for their SMTP credentials that would need to switch to a different service if there isn't oAuth support available, which would be a lot of overhead for some of them.

**Reply by brandonkelly** [reply]
We’re keeping an eye on https://github.com/symfony/symfony/pull/52546. It looks like that will get pulled in for Symfony 7.2.

**Comment by alexanderbuergin** [comment]
@brandonkelly
Is there any update on this topic?

Microsoft will discontinue support for basic SMTP authentication by September 2025 (see [SendLayer article](https://sendlayer.com/blog/microsoft-smtp-auth-changes-fix-email/#:~:text=Microsoft%20is%20phasing%20out%20Basic,discontinued%20completely%20by%20September%202025.)).

This affects many Craft CMS users who need to send emails via Microsoft 365. oAuth will become mandatory, and relying on external SMTP services is not a practical long-term solution for most clients.

Questions:
– Is there any news on if or when a dedicated oAuth transport will be implemented in Craft CMS?
– Do you already know how Microsoft’s change will specifically affect Craft CMS installations?
– Can we expect a solution before the deadline?

Thank you for a short update.

**Reply by brandonkelly** [reply]
I don’t have an update, sorry. We continue to monitor https://github.com/symfony/symfony/pull/52546 and https://github.com/symfony/symfony/issues/48888.

**Reply by alexanderbuergin** [reply]
Thanks @brandonkelly.

**Comment by funkysoul** [comment]
@brandonkelly any updates on this matter? getting more and more requests of clients where their IT are switching to oauth.. :-/

**Reply by alexanderbuergin** [reply]
True. We had 4 asks for that in the last 60 Days.

**Comment by brandonkelly** [comment]
We’re continuing to watch https://github.com/symfony/symfony/pull/52546, but in the meantime, a Microsoft 365 mail transporter was just added to the Plugin Store: https://plugins.craftcms.com/microsoft-365-mail-transport

**Comment by superflausch** [comment]
This is also relevant for some of our projects, so here’s a quick update from what I’ve seen:
It looks like this will be included in the upcoming Symfony 7.4 release (https://github.com/symfony/symfony/pull/61290), which is planned for November (https://symfony.com/releases/7.4). I’d assume it will be adopted into CraftCMS after that.

In the meantime, we’ll likely need to rely on the plugin for clients who switch before the release.

**Comment by remoflury** [comment]
@brandonkelly is there an update regarding a timeline for the integration of MS oauth into craft? Thanks in advance.

**Reply by brandonkelly** [reply]
We have a plugin that is ready to go, but we are waiting on symfony-mailer 7.4 and microsoft-graph-mailer 7.4 to be released.

**Reply by darylknight** [reply]
As far as I can tell, both of these packages are released now. Would you please update here once this plugin is available? I have a client who wants to get moving as Microsoft are going to start shutting down this functionality in Q1 next year.

**Comment by brandonkelly** [comment, marked as answer]
We just released a new [Microsoft Graph Mailer](https://plugins.craftcms.com/microsoft-graph-mailer?craft5) plugin, which adds a [Microsoft Graph](https://learn.microsoft.com/en-us/graph/overview) mailer integration 🎉

**Reply by nelhop** [reply]
Is it planned to make this plugin also compatible with Craft 4?

**Reply by brandonkelly** [reply]
No, we’re no longer officially supporting Craft 4.
